src/components/calculator.js

```diff
@@ -1,107 +1,214 @@
-        <button className="calc-btn bg-light-gray" type="button">
-          3
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          x
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          4
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          5
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          6
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          -
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          7
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          8
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray" type="button">
-          9
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          +
-        </button>
-      </div>
-      <div className="grid-item zero-btn">
-        <button className="calc-btn bg-light-gray" type="button">
-          0
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-light-gray text-bold" type="button">
-          .
-        </button>
-      </div>
-      <div className="grid-item">
-        <button className="calc-btn bg-brown" type="button">
-          =
-        </button>
-      </div>
-    </div>
-  </section>
-);
+function Calculator() {
+  const [oldState, newState] = useState({
+    total: 0,
+    next: null,
+    operation: null,
+  });
+  const handleClick = (e) => {
+    newState((oldState) => calculate(oldState, e.target.name));
+  };
+
+  return (
+    <section>
+      <div className="grid-container">
+        <CalcScreen answer={oldState} />
+        <div className="grid-item">
+          <button
+            onClick={handleClick}
+            name="AC"
+            className="calc-btn bg-light-gray"
+            type="button"
+          >
+            AC
+          </button>
+        </div>
+        <div className="grid-item">
+          <button
+            className="calc-btn bg-light-gray"
+            onClick={handleClick}
+            name="-"
+            type="button"
+          >
+            +/-
+          </button>
+        </div>
+        <div className="grid-item">
+          <button
+            className="calc-btn bg-light-gray"
+            onClick={handleClick}
+            name="%"
+            type="button"
+          >
+            %
```

Comment:
- [ ] Kindy remove unused codes and files from your codebase as this liters and slows than your app for no reason. Files like, reportWebVitalis.js, setupTests.js, logo.svg, App.test.js, App.css and some I've failed to mention.